### PR TITLE
Upgrade the Gradle wrapper to 7.6.1

### PR DESCRIPTION
**Pull request by dmikurube**
Along with the Gradle wrapper,
- Upgrade the "org.embulk.embulk-plugins" Gradle plugin to 0.6.1
- Use Gradle's Java toolchain
- Add "checkDependencies" task in build.gradle
- Stop releasing Gradle's ".module" file
- Upgrade the base Embulk SPI to v0.11
- Upgrade embulk-util-config to 0.3.4
- Upgrade embulk-util-timestamp to 0.2.2
- Upgrade jackson-databind to 2.6.7.5

### CODE REVIEW

**Comment by dmikurube** [comment]
It turned out that msgpack-core's `MessageUnpacker` unpacks some MessagePack integers within the Java `long` range into `ImmutableBigIntegerValueImpl`.

Top-level MessagePack integers are handled properly in:
https://github.com/embulk/embulk-parser-msgpack/blob/v0.4.0/src/main/java/org/embulk/parser/msgpack/MsgpackParserPlugin.java#L474-L487

However, MessagePack integers inside an Array or a Map had been conveyed as-is, without converting into `long`. `ArrayValue` or `MapValue` might include `ImmutableBigIntegerValueImpl`.

Then, the conveyed `ImmutableBigIntegerValueImpl` might be passed to the new Embulk SPI `JsonLong`, and it was rejected.
https://github.com/embulk/embulk-spi/blob/v0.11/src/main/java/org/embulk/spi/json/JsonLong.java#L45-L51

The test failure in 6f72988b954b2ed47821af837bef529bcf851029 is because of this. It wouldn't work with Embulk v0.10.42+ that introduced `JsonValue`.

----

`JsonLong` could be "fixed" like below, but updating the SPI is not very trivial. This fix would be merged at least in SPI v1.0, but we have not determined whether we release SPI v0.12.
https://github.com/embulk/embulk-spi/pull/4

Unpacking MessagePack directly into the new `JsonValue` would be simple, but then it must require Embulk v0.10.42+ immediately. It means that users may have to upgrade the Embulk core and this plugin at the same time, not one-by-one. We wanted to avoid this situation as far as possible.
https://github.com/embulk/embulk/releases/tag/v0.10.42

In order to make it available to one-by-one upgrade,
* We are going to release `embulk-parser-msgpack` v0.5.0, which still calls MessagePack `Value`-based Embulk SPIs, while converting all `ImmutableBigIntegerValueImpl` into `ImmutableLongValueImpl` even inside arrays and maps. It would work both with Embulk v0.10.41 (or earlier) and v0.10.42+.
* Next, we are going to release `embulk-parser-msgpack` v0.6.0, which calls the new `JsonValue`-based Embulk SPIs. It would work only with Embulk v0.10.42+.

By this strategy, users can do upgrades one-by-one:
1. Upgrade `embulk-parser-msgpack` to v0.5.0 first, even with Embulk v0.10.41 or earlier.
2. Upgrade the Embulk core to v0.10.42+ or v0.11.
3. Upgrade `embulk-parser-msgpack` to v0.6.0.

File: src/main/java/org/embulk/parser/msgpack/MsgpackParserPlugin.java [hunk 1]
```diff
@@ -627,6 +634,73 @@ public int compare(MessageBuffer o1, MessageBuffer o2)
         }
     }
 
+    private static Value repackValueInLongRange(final Value value) {
```
Comment (hiroyuki-sato) [inline src/main/java/org/embulk/parser/msgpack/MsgpackParserPlugin.java:637]: A comment(why we need to repack Value) may help.
https://github.com/embulk/embulk-parser-msgpack/pull/27#issuecomment-1767658946

**Review by hiroyuki-sato — APPROVED** [review]
LGTM👍

One minor question.

`./gradlew dependencies` outputs below.


```
+--- org.embulk:embulk-spi:0.11
|    +--- org.slf4j:slf4j-api:2.0.7
|    \--- org.msgpack:msgpack-core:0.8.24
```

will change to `org.embulk:msgpack-core` instead of `org.msgpack:msgpack-core` in  future release?

File: src/main/java/org/embulk/parser/msgpack/MsgpackParserPlugin.java (hunk 1 above)
Comment (dmikurube) [reply src/main/java/org/embulk/parser/msgpack/MsgpackParserPlugin.java:637]: Yeah, thought about that, but they will go away immediately very soon in v0.6.0 when moving to `JsonValue`. They're only for this version v0.5.0.

**Comment by dmikurube** [comment]
> will change to `org.embulk:msgpack-core` instead of `org.msgpack:msgpack-core` in future release?

Yes, planned to do so in v0.6.0 when migrating to `JsonValue`.

**Comment by dmikurube** [comment]
Thanks!

**Pull request merged**